From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Praschl <praschlpaul@g-p.at>
Date: Sun, 26 Jul 2026 17:27:04 +0200
Subject: [PATCH 048/117] arm64: defconfig: enable Apple J172 Wi-Fi support

Build the H9P, Apple Wi-Fi pwrctrl, T8010 WLAN supplier, D2365 GPIO
and SysCfg layout drivers into the arm64 reference configuration.

Keep DART and reserved-memory NVMEM available during the built-in PCIe
probe path.

Signed-off-by: Paul Praschl <praschlpaul@g-p.at>

Orchard-replay-of: 2123b37bfa08d61ffd044ee2ce65fab351ed1c7b
---
 arch/arm64/configs/defconfig | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/configs/defconfig b/arch/arm64/configs/defconfig
index ac1439b91..a9a7a17d1 100644
--- a/arch/arm64/configs/defconfig
+++ b/arch/arm64/configs/defconfig
@@ -1622,7 +1622,7 @@ CONFIG_ARM_SMMU_V3=y
 CONFIG_QCOM_IOMMU=y
 CONFIG_ROCKCHIP_IOMMU=y
 CONFIG_TEGRA_IOMMU_SMMU=y
-CONFIG_APPLE_DART=m
+CONFIG_APPLE_DART=y
 CONFIG_MTK_IOMMU=y
 CONFIG_REMOTEPROC=y
 CONFIG_IMX_REMOTEPROC=y
@@ -1839,7 +1839,7 @@ CONFIG_NVMEM_MESON_EFUSE=m
 CONFIG_NVMEM_MTK_EFUSE=y
 CONFIG_NVMEM_QCOM_QFPROM=y
 CONFIG_NVMEM_QCOM_SEC_QFPROM=m
-CONFIG_NVMEM_RMEM=m
+CONFIG_NVMEM_RMEM=y
 CONFIG_NVMEM_ROCKCHIP_EFUSE=y
 CONFIG_NVMEM_ROCKCHIP_OTP=y
 CONFIG_NVMEM_SNVS_LPGPR=y
@@ -1980,3 +1980,8 @@ CONFIG_CORESIGHT_STM=m
 CONFIG_CORESIGHT_CPU_DEBUG=m
 CONFIG_CORESIGHT_CTI=m
 CONFIG_MEMTEST=y
+CONFIG_PCIE_APPLE_H9P=y
+CONFIG_PCI_PWRCTRL_APPLE_WIFI=y
+CONFIG_APPLE_T8010_WLAN=y
+CONFIG_GPIO_APPLE_D2365=y
+CONFIG_NVMEM_LAYOUT_APPLE_SYSCFG=y
